extravaganza/2016-03-March.md

```diff
@@ -0,0 +1,85 @@
+meeting is open to the public; you should stop by!
+
+You can check out the [wiki page][wiki] that we use to organize the meeting, or
+view a [recording of the meeting][recording] in Air Mozilla. Or just read on for
+a summary!
+
+[wiki]: https://wiki.mozilla.org/Webdev/Meetings/2016/March_1
+[recording]: https://air.mozilla.org/webdev-extravaganza-march-2016/
+
+## Shipping Celebration
+The shipping celebration is for anything we finished and deployed in the past
+month, whether it be a brand new site, an upgrade to an existing one, or even a
+release of a library.
+
+### Peep 3.1.1
+First up was [ErikRose][] with news of a new release of [Peep][]. The new
+version fixes the `peep port` command so that it correctly emits URL-based
+requirements.
+
+Erik also wants to remind everyone to drop Peep and upgrade to [Pip 8][].
+
+[ErikRose]: https://mozillians.org/en-US/u/ErikRose/
+[Peep]: https://github.com/erikrose/peep/
+[Pip 8]: https://pip.pypa.io/en/stable/
+
+### SHIELD Recipe Server
+Next up was [mythmon][], who talked about the [Recipe Server][], a system to
+help Firefox respond to user issues and test new features quickly and easily.
+The development server for the service is now live, and is deployed to AWS
+using Docker and gunicorn.
+
+[mythmon]: https://mozillians.org/en-US/u/mythmon/
+[Recipe Server]: https://wiki.mozilla.org/Firefox/Recipe_Server
+
+### Go Faster System Addon Update
+Next we heard from [laura][] who shared the news that the first [Go Faster][]
+system addon update shipped! System addons are addons that come bundled with
+Firefox, but are updated outside of the normal 6-8 week release process. They
+enable us to try new features out and respond to user feedback quicker than
+before.
+
+[laura]: https://mozillians.org/en-US/u/lthomson/
+[Go Faster]: https://wiki.mozilla.org/Firefox/Go_Faster
+
+## Roundtable
+The Roundtable is the home for discussions that don't fit anywhere else.
+
+### Debugging Python with gdb
+ErikRose next talked about how he dealt with a segfault in [DXR][]'s indexing
+job. The end result of his trial is a description of
+[how to debug Python stack frames using gdb on Ubuntu Trusty][gdb].
+
+[DXR]: https://dxr.mozilla.org/
+[gdb]: https://github.com/erikrose/status/wiki/GdbWithPythonOnUbuntu
+
+### Conditional Requirements in Python are Bad
+Erik also shared a story about conditional requirements in Python. Because
+Python uses executable Python files for describing the requirements for
+packages, some projects change their requirements list based on the environment.
+For example, some projects detect whether they're running in Python 2.6 or 2.7
+and include libraries that implement features that may be missing in Python 2.
+
+The problem is that pip caches packages _after_ evaluating `setup.py` files,
+meaning that a cached package built for Python 2 may be used when installing in
+Python 3. The solution is to use a feature called [environment markers][] that
```

Comment:
Actually not—I believe Python 2 and 3 are kept separate. However, I know for a fact that 2.6 and 2.7 get mixed up. We might also hedge by saying "In some recent versions of pip…", because I think there's at least a bug open about this, if not a patch merged.